First release

nicola-calonaci opened the pull request.

valerianilucrezia committed:
add modules.json

valerianilucrezia committed:
fix last 2 test

valerianilucrezia committed:
fix pre-commit

valerianilucrezia committed:
change usage

valerianilucrezia committed:
minor

valerianilucrezia committed:
adjust pre-commit p1

valerianilucrezia committed:
fix pre-commit final

valerianilucrezia committed:
minor in docs

valerianilucrezia committed:
minor schema

valerianilucrezia committed:
minor in test config

valerianilucrezia committed:
remove check_max, + minors

valerianilucrezia committed:
add params description of mobster/viber/pyclone + modify test

valerianilucrezia committed:
add test_full

elena-buscaroli committed:
updating docker image and adding version.yml

valerianilucrezia committed:
add ctree, TINC, joinCNAqc

elena-buscaroli committed:
adding version.yml to pyclone-vi

valerianilucrezia committed:
add yml and modify config

valerianilucrezia committed:
solve out problem

kdavydzenka committed:
update docker tag version + adding version.yml

kdavydzenka committed:
adding SparseSig, Sigprofiler params

kdavydzenka committed:
adding SparseSig, Sigprofiler params

kdavydzenka committed:
uniform dir + minor updates

elena-buscaroli committed:
fix versions.yml in subclonal deconvolution

kdavydzenka committed:
fix versions yml

nf-core-bot committed:
Template update for nf-core/tools version 3.1.2

kdavydzenka committed:
fix yml version

giorgiagandolfi committed:
fix typo in annotate_driver

kdavydzenka committed:
adding version yml

kdavydzenka committed:
adding version yml

kdavydzenka committed:
fix trailing whitespace SigProfiler

giorgiagandolfi committed:
adding version.yml to tinc

giorgiagandolfi committed:
fix sigprofiler module + mc subworkflow

valerianilucrezia committed:
update ensemblvep.config

giorgiagandolfi committed:
fix sigprofiler output dir + versions.yml

giorgiagandolfi committed:
finalize output.md

elena-buscaroli committed:
Add resources selectors

elena-buscaroli committed:
fixing trailing whitespaces

elena-buscaroli committed:
nf-core modules patch

nf-core-bot committed:
Template update for nf-core/tools version 3.2.0

elena-buscaroli committed:
updating resources labels

nicola-calonaci committed:
Merge pull request #24 from caravagnalab/dev

Release checklist completed

nicola-calonaci committed:
Merge branch 'dev' into nf-core-template-merge-3.2.0

nicola-calonaci committed:
Merge pull request #23 from nf-core/nf-core-template-merge-3.2.0

Important! Template update for nf-core/tools v3.2.0

nicola-calonaci committed:
Fixed plugin typo

elena-buscaroli committed:
removing obsolete modules.config

elena-buscaroli committed:
updating test.config params

elena-buscaroli committed:
update modules.json removing old nf-core modules

elena-buscaroli committed:
adding templates

elena-buscaroli committed:
deleting auxiliary scripts included in templates

elena-buscaroli committed:
deleting auxiliary scripts included in templates

elena-buscaroli committed:
updating docker containers and setting templates

elena-buscaroli committed:
update nextflow.config

nicola-calonaci committed:
Merge pull request #25 from caravagnalab/dev

Docker and script templates

elena-buscaroli committed:
first release checks

valerianilucrezia committed:
fix pre-commit

nicola-calonaci committed:
Merge pull request #26 from caravagnalab/dev

First release final fixes

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
+It is recommended to use the absolute path of the files, but a relative path should also work.
 
-### Updating the pipeline
+For the joint analysis of multiple samples, a tumour BAM file is required for each sample, such that the number of reads of a private mutation can be retrieved for all the samples thorugh `mpileup`.
```
Does it take cram files?

nicola-calonaci replied:
It does indeed. We have updated the documentation accordingly.

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
 
-> We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when this is not possible, Conda is also supported.
+The tumourevo pipeline supports variant annotation, driver annotation, quality control processes, subclonal deconvolution and signature deconvolution analysis through various tools. It can be used to analyse both single sample experiments and longitudinal/multi-region assays, in which multiple samples of the same patient are avaiable.
+As input, you must provide at least information on the samples, the VCF file from one of the supported callers and the output of one of the supported copy number caller. By default, if multiple samples from the same patient are provided, they will be analysed in a multivariate framework (which affects in particular the subclonal deconvolution deconvolution steps) to retrieve information useful in ...
```
```suggestion
As input, you must provide at least information on the samples, the VCF file from one of the supported callers and the output of one of the supported copy number callers. By default, if multiple samples from the same patient are provided, they will be analysed in a multivariate framework (which affects in particular the subclonal deconvolution steps) to retrieve information useful in the reconstruction of the evolutionary process. Depending on the variant calling strategy (single sample or multi sample) and the provided input files, different strategies will be applied.
```

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
 
-The pipeline also dynamically loads configurations from [https://github.com/nf-core/configs](https://github.com/nf-core/configs) when it runs, making multiple config profiles for various institutional clusters available at run time. For more information and to see if your system is available in these configs please see the [nf-core/configs documentation](https://github.com/nf-core/configs#documen ...
+<!-- aggiungi un riassunto di cosa voglia dire single e multi sample (analisi multivariata, soprattutto per subclonal deconv)
+E' possibile usarla sia nel caso di vc multi sample che indipendente -->
```
Non posso leggerlo

nicola-calonaci replied:
Abbiamo rimosso i commenti in Italiano :)

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
-* `test`
-  * A profile with a complete configuration for automated testing
-  * Includes links to test data so needs no other parameters
+<!-- If the variant calling had been performed indepentently on each sample from the same patient,
```
This is commented out?

nicola-calonaci replied:
Removed.

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
+##### 2. Single sample variant calling
 
-### `--single_end`
+If the variant calling performed independently on each sample, even if coming from the same patient, you can provide the BAM and BAI files from each tumour sample. In this way, a classical pileup strategy will be used in order to retrieve the depth for all samples of private mutations, in order to correctly perform the subclonal deconvolution analysis.
```
Think this is what you mean here.
```suggestion
If the variant calling has been performed independently on each sample, even if coming from the same patient, you can provide the BAM and BAI files from each tumour sample. In this way, a classical pileup strategy will be used in order to retrieve the depth for all samples of private mutations, in order to correctly perform the subclonal deconvolution analysis.
```

nicola-calonaci replied:
Updated.

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
 
-### `--genome` (using iGenomes)
+During the QC step, the pipeline will combine purity, copy number and mutation data to perform quality control on the copy number calls, by applying the [CNAqc algorithm](https://caravagnalab.github.io/CNAqc/). Each segment (for each sample) will be flagged as passing or not the QC, in the given combination of estimated purity and ploidy. According to CNAqc, a badly called segment should be recal ...
+After CNAqc quality control, all the segments (coming from samples of the same patient) are used to build a multi-sample CNAqc object, in which a common segmentation is applied. In this way, only those regions that are shared among all samples will be kept in the new object. It is possible to control wheter to include or not in the new segmentation the segments, for each sample, that do not pass  ...
```
```suggestion
After CNAqc quality control, all the segments (coming from samples of the same patient) are used to build a multi-sample CNAqc object, in which a common segmentation is applied. In this way, only those regions that are shared among all samples will be kept in the new object. It is possible to control whether to include or not in the new segmentation the segments, for each sample, that do not pass the QC test using the `--filter` flag. If this is set to true, only QC passing segments for each sample will be used to build the mCNAqc object and will then be passed to the subclonal deconvolution steps. This will lead to exclude some regions of the genome and the mutations that sit on it, but should result in more precise analyses. Otherwise, keeping the segments that do not pass the QC will result in not losing any mutations, but might lead to less precise results in the subclonal deconvolution steps.
```

nicola-calonaci replied:
Updated.

SPPearce commented on CITATIONS.md:
This hasn't been updated at all.

nicola-calonaci replied:
Thanks, we have updated the file.

SPPearce commented on README.md:
```diff
@@ -1,77 +1,125 @@
 ## Introduction
 
-The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It comes with docker containers making installation trivial and results highly reproducible.
+**nf-core/tumourevo** is a bioinformatics pipeline to model tumour evolution from whole-genome sequencing (WGS) data. The pipeline performs state-of-the-art downstream analysis of variant and copy-number calls from tumour-normal matched sequecing assays, reconstructing the evolutionary processes leading to the observed tumour genome. This analysis can be done at the level of single samples, multi ...
```
```suggestion
**nf-core/tumourevo** is a bioinformatics pipeline to model tumour evolution from whole-genome sequencing (WGS) data. The pipeline performs state-of-the-art downstream analysis of variant and copy-number calls from tumour-normal matched sequencing assays, reconstructing the evolutionary processes leading to the observed tumour genome. This analysis can be done at the level of single samples, multiple samples from the same patient (multi-region/longitudinal assays), and of multiple patients from distinct cohorts.
```

nicola-calonaci replied:
Updated.

SPPearce commented on assets/methods_description_template.yml:
Also not updated yet.

vvvirgy replied:
We do not have a published paper for the pipeline yet. Can we just leave this as it is and modify it when we will have it?

SPPearce replied:
You don't need the published paper, but this is also where you link to the tools that you use.
Having said that, I see a number of other pipelines including sarek don't, so this is probably fine.

nicola-calonaci replied:
OK, thanks! We also noticed that the tools are already linked in the https://github.com/caravagnalab/tumourevo/blob/dev/CITATIONS.md.

SPPearce commented on assets/samplesheet.csv:
Can you update this as an example for your pipeline?
Or delete it, it is just unhelpful at the moment.

nicola-calonaci replied:
Thanks, we have deleted the samplesheet file.

SPPearce commented on assets/schema_input.json:
```diff
@@ -0,0 +1,194 @@
+                "exists": true
+            },
+            "tumour_alignment": {
+                "errorMessage": "Tumour BAM or CRAM file cannot contain spaces and must have extension '.bam' or '.cram'",
```
Ah, so we do allow cram files.

nicola-calonaci replied:
Yes, cram files are allowed.

SPPearce commented on conf/base.config:
```diff
@@ -1,51 +1,56 @@
-  memory = { check_max( 7.GB * task.attempt, 'memory' ) }
-  time = { check_max( 4.h * task.attempt, 'time' ) }
+    errorStrategy = { task.exitStatus in ((130..145) + 104) ? 'retry' : 'finish' }
+    maxRetries    = 1
```
This will only retry once, whereas something above suggested twice.

nicola-calonaci replied:
This is the nf-core default parameterisation, that's why we left it s it is. However, all the modules in the pipeline are labelled so that every type of process has its own error strategy and number of retries.

SPPearce commented on conf/modules/driver_annotation.config:
```diff
@@ -0,0 +1,12 @@
+process{
+    withName: 'ANNOTATE_DRIVER' {
+        errorStrategy = 'finish'
```
Why the different error strategies between different modules?

nicola-calonaci replied:
All modules now follow the 'retry' error strategy, except for ctree, which has 'ignore' because the only cases with error are expected due to tool limitations, and it does not affect the correct functioning of the workflow.

SPPearce commented on conf/modules/lifter.config:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    withName: 'BCFTOOLS_MPILEUP' {
+        ext.args = { "--annotate FORMAT/AD,FORMAT/ADF,FORMAT/ADR,FORMAT/DP,FORMAT/SP,INFO/AD,INFO/ADF,INFO/ADR --threads 36" }
```
Is this setting the cpus used to be 36? Because that'll be a problem on some systems.
This should probably be being set inside the nf-core module itself, but the help file suggests:

> --threads INT
> Use multithreading with INT worker threads. The option is currently used only for the compression of the output stream, only when --output-type is b or z. Default: 0.
Probably therefore wants to be on args3 not args or args2.

nicola-calonaci replied:
Following standard nf-core/sarek, we removed the string from the `ext.args` and `ext.args2`. This way module now uses its own default `--threads` parameter.

SPPearce commented on conf/modules/mobsterh.config:
```diff
@@ -0,0 +1,30 @@
+process {
+    withName: 'MOBSTERh' {
```
Why does this have a lowercase h? We generally use all upper case for module names.

nicola-calonaci replied:
Updated.

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
+| `cna_segments`           | Full path to the segmentation files and copy number state from copy-number calling. <br /> _Required_                                                                                                                                     |
+| `cna_extra`              | Full path to files including the ploidy and purity estimate from the copy-number caller. <br /> _Required_                                                                                                                                |
+| `cancer_type`            | Tumour type (either `PANCANCER` or one of the tumour type present in the driver table) <br /> _Required_                                                                                                                                  |
+| `tumour_alignment`       | Full path to the tumour bam file. <br /> _Optional_                                                                                                                                                                                       |
```
Or cram file

nicola-calonaci replied:
Updated.

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,342 @@
+| `cna_extra`              | Full path to files including the ploidy and purity estimate from the copy-number caller. <br /> _Required_                                                                                                                                |
+| `cancer_type`            | Tumour type (either `PANCANCER` or one of the tumour type present in the driver table) <br /> _Required_                                                                                                                                  |
+| `tumour_alignment`       | Full path to the tumour bam file. <br /> _Optional_                                                                                                                                                                                       |
+| `tumour_alignment_index` | Full path to the tumour bam index file. <br /> _Optional_                                                                                                                                                                                 |
```
Or crai file

nicola-calonaci replied:
Updated.

SPPearce commented on modules/local/CNAqc2tsv/main.nf:
```diff
@@ -0,0 +1,17 @@
+process RDS_PROCESSING {
```
Why is this called RDS_PROPCESSING inside a file called CNAqc2tsv?

nicola-calonaci replied:
Thanks for this comment. The process is now called CNAQC2TSV, accordingly.

SPPearce commented on modules/local/SigProfiler/SigProfiler/main.nf:
```diff
@@ -0,0 +1,169 @@
+    tag "$meta.id"
+    label "process_high"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://katiad/sigprofiler:version1.0' :
```
This will require singularity users to pull and convert the docker container, which in my experience fails half the time.

nicola-calonaci replied:
Now supports conda and seqera containers (module now on nf-core)

SPPearce reviewed (changes requested):
I don't have time to go through the code, but I have a few comments on the documentation part.
You have a number of local modules that could be put into nf-core/modules.
Does the pipeline support conda? I don't see any environment files for the local modules, they all seem to use custom docker containers. 
I suspect that these could be replaced by conda environments/seqera containers, which would also make things better for singularity/apptainer uses.

You seem to be exposing every single possible parameter to your scripts as parameters, is this really necessary?
This is leading to some very bloated expressions inside your local R modules.

vvvirgy committed:
documentation typos and input csv fixed

vvvirgy committed:
fixing documentation

valerianilucrezia committed:
minor vcf_parser vep field

giorgiagandolfi committed:
minor changes vcf2CNAqc

giorgiagandolfi committed:
mc mobster template script

giorgiagandolfi committed:
mc viber template script

nf-core-bot committed:
Template update for nf-core/tools version 3.2.1

valerianilucrezia committed:
minor config

valerianilucrezia committed:
minor to VIBER - solve opt error

kdavydzenka committed:
minor updates sparsesignatures save mutCounts data

nf-core-bot committed:
Template update for nf-core/tools version 3.3.1

valerianilucrezia committed:
add bcftools filtering + change parser

valerianilucrezia committed:
minor

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates signature

kdavydzenka committed:
minor updates Sigprofiler: DBS and ID fit

kdavydzenka committed:
minor updates Sigprofiler: DBS and ID fit

kdavydzenka committed:
minor fix sigprofiler

valerianilucrezia committed:
minor sigprofiler

valerianilucrezia committed:
modify driver filter

valerianilucrezia committed:
minor sparcsesignature main_script.R

valerianilucrezia committed:
minor driver

nf-core-bot committed:
Template update for nf-core/tools version 3.3.2

kdavydzenka committed:
minor update Sigprofiler

valerianilucrezia committed:
add support to Battenberg calls

valerianilucrezia committed:
add nf-core modules

valerianilucrezia committed:
correct lint

valerianilucrezia committed:
minor

valerianilucrezia committed:
new nf module

valerianilucrezia committed:
remove unused params

valerianilucrezia committed:
solve lint

valerianilucrezia committed:
minor to docs

valerianilucrezia committed:
minor nextflow.config

valerianilucrezia committed:
solve lint

valerianilucrezia committed:
solve issue with nextflow.config

valerianilucrezia committed:
Merge pull request #35 from caravagnalab/dev

Update tumourevo with nf-core modules

valerianilucrezia committed:
add new container and yml

valerianilucrezia committed:
minor lint

valerianilucrezia committed:
prettier check

valerianilucrezia committed:
add conda support

valerianilucrezia committed:
solve test_full and change label

valerianilucrezia committed:
Merge pull request #36 from caravagnalab/dev

Add support to conda and modify test_full

nf-core-bot committed:
Template update for nf-core/tools version 3.4.1

valerianilucrezia committed:
add last nf-core modules

valerianilucrezia committed:
update pipeline schema

valerianilucrezia committed:
vep patch

valerianilucrezia committed:
Merge pull request #38 from caravagnalab/dev

Add last nf-core modules:
- VIBER
- CNAqc
- pyclone-vi

valerianilucrezia committed:
solve vep error

valerianilucrezia committed:
Merge pull request #39 from caravagnalab/dev

Solve vep error

elena-buscaroli committed:
minor updates to documentation

giorgiagandolfi committed:
fix process name in formatter local modules/subworkflow

vvvirgy committed:
updating citations and removing example samplesheet

vvvirgy committed:
fixing CITATIONS.md typo

elena-buscaroli committed:
minor updates to documentation and config

elena-buscaroli committed:
adding yml in conf patterns

elena-buscaroli committed:
minor changes

elena-buscaroli committed:
Fixing failing pre-commit

elena-buscaroli committed:
Merge branch 'nf-core:dev' into dev

nicola-calonaci committed:
Merge pull request #40 from caravagnalab/dev

FInal updates for first release

SPPearce commented on CITATIONS.md:
```diff
@@ -10,13 +10,49 @@
+
+  > Gillis, S., Roth, A. PyClone-VI: scalable inference of clonal population structures using whole genome data. BMC Bioinformatics 21, 571 (2020). https://doi.org/10.1186/s12859-020-03919-2
+
+- [SigProfiler](https://www.sciencedirect.com/science/article/pii/S2666979X22001240?via%3Dihub)
```
```suggestion
- [SigProfiler](https://www.sciencedirect.com/science/article/pii/S2666979X22001240)
```

SPPearce commented on CITATIONS.md:
```diff
@@ -10,13 +10,49 @@
+
+  > Islam SMA, Díaz-Gay M, Wu Y, Barnes M, Vangara R, Bergstrom EN, He Y, Vella M, Wang J, Teague JW, Clapham P, Moody S, Senkin S, Li YR, Riva L, Zhang T, Gruber AJ, Steele CD, Otlu B, Khandekar A, Abbasi A, Humphreys L, Syulyukina N, Brady SW, Alexandrov BS, Pillay N, Zhang J, Adams DJ, Martincorena I, Wedge DC, Landi MT, Brennan P, Stratton MR, Rozen SG, and Alexandrov LB (2022) Uncovering nov ...
+
+- [SparseSignatures](https://www.sciencedirect.com/science/article/pii/S2666166722003938?via%3Dihub)
```
```suggestion
- [SparseSignatures](https://www.sciencedirect.com/science/article/pii/S2666166722003938)
```

nicola-calonaci replied:
updated

SPPearce commented on conf/modules/cnaqc.config:
Why have you added yml to all these? Is it to publish the versions.yml file? Because if so we don't generally publish those, they are only used to collect the versions prior to multiqc (and won't be required once we move to topic channels).

nicola-calonaci replied:
Sorry, we thought these files were required as for nf-core guidelines. We have now removed them form the list of files to be published.

SPPearce commented on modules/local/cnaqc2tsv/templates/main_script.R:
```diff
@@ -0,0 +1,94 @@
+        # check if the original cnaqc obj exist
+        check_or = any(names(m_cnaqc_obj) == type)
+        if (check_or == FALSE) {
+            cli::cli_abort(c("mCNAqc object was build without keeping original CNAqc objects"),
```
```suggestion
            cli::cli_abort(c("mCNAqc object was built without keeping original CNAqc objects"),
```

nicola-calonaci replied:
We have removed this piece of code, since it was deprecated

nicola-calonaci committed:
Merge branch 'dev' into nf-core-template-merge-3.4.1

SPPearce commented on modules/local/cnaqc2tsv/templates/main_script.R:
```diff
@@ -0,0 +1,94 @@
+for (s in get_sample_name(multi_cnaqc)){
+    purity = mutations_multisample[[s]][["purity"]]
+    multisample_jointTable[[s]] = mutations_multisample[[s]][["mutations"]] %>%
+        dplyr::mutate(purity=purity)
```
Is this not just this:
```suggestion
    multisample_jointTable[[s]] = mutations_multisample[[s]] %>% 
      dplyr::select(mutations, purity)
```

Or you could use purrr::map (or friends) rather than use the `for` loop.

nicola-calonaci replied:
updated

SPPearce commented on modules/local/get_positions/main.nf:
```diff
@@ -0,0 +1,41 @@
+
+    output:
+    tuple val(meta), path("*_all_positions.rds"),   emit: all_pos
+    path "versions.yml",                            emit: versions
```
```suggestion
    tuple val(meta), path("*_all_positions.rds"), emit: all_pos
    path "versions.yml",                          emit: versions
```

nicola-calonaci replied:
updated

SPPearce commented on modules/local/get_positions/main.nf:
```diff
@@ -0,0 +1,41 @@
+
+    positions = lapply(strsplit("$rds_list", " ")[[1]], FUN = function(rds){
+        df = readRDS(rds)
+        df = df[[1]]\$mutations %>% dplyr::mutate(id = paste(chr, from, to, sep = ":")) %>% dplyr::select(chr, from, to, ref, alt, id)
```
```suggestion
        df = df[[1]]\$mutations %>% 
          dplyr::mutate(id = paste(chr, from, to, sep = ":")) %>% 
          dplyr::select(chr, from, to, ref, alt, id)
```

nicola-calonaci replied:
updated

SPPearce commented on modules/local/get_positions/main_rel.nf:
```diff
@@ -0,0 +1,48 @@
+
+    output:
+    tuple val(meta), path("*_positions_missing"),   emit: bed
+    path "versions.yml",                            emit: versions
```
```suggestion
    tuple val(meta), path("*_positions_missing"), emit: bed
    path "versions.yml",                          emit: versions
```

nicola-calonaci replied:
updated

vvvirgy committed:
removing outdated modules tests

vvvirgy committed:
Merge branch 'dev' of https://github.com/caravagnalab/tumourevo into dev

nicola-calonaci committed:
Merge pull request #41 from caravagnalab/dev

removing outdated tests

Commit:
Merge branch 'dev' into nf-core-template-merge-3.4.1

Commit:
merging new template

Commit:
ro create

Commit:
modules and workflows updated

Commit:
modules and workflows updated

Commit:
round on linting

Commit:
round on linting

Commit:
round on linting

Commit:
round on linting

SPPearce commented on conf/base.config:
```diff
@@ -1,51 +1,60 @@
+    // See https://www.nextflow.io/docs/latest/config.html#config-process-selectors
+    withLabel:process_single {
+        cpus   = { 1                   }
+        memory = { 20.GB * task.attempt }
```
That is a LOT to be giving to every single process?

SPPearce replied:
This is still asking for 20Gb for tasks marked as single?

SPPearce commented on conf/base.config:
```diff
@@ -1,51 +1,60 @@
+    }
+    withLabel:process_low {
+        cpus   = { 4     * task.attempt }
+        memory = { 50.GB * task.attempt }
```
50GB for low?

SPPearce commented on modules/local/annotate_driver/main.nf:
```diff
@@ -0,0 +1,83 @@
+
+    output:
+    tuple val(meta), path("*.rds"),     emit: rds
+    path "versions.yml",                emit: versions
```
```suggestion
    tuple val(meta), path("*.rds"), emit: rds
    path "versions.yml",            emit: versions
```

nicola-calonaci replied:
updated

SPPearce commented on modules/local/join_cnaqc/main.nf:
```diff
@@ -0,0 +1,66 @@
+    def args                                = task.ext.args
+    def prefix                              = task.ext.prefix                                       ?: "${meta.id}"                                      ?: ''
+    def qc_filter                           = args!='' && args.qc_filter                            ?  "$args.qc_filter" : ""
+    def keep_original                       = args!='' && args.keep_original                        ?  "$args.keep_original" : ""
```
Can remove quite a bit of this space.

nicola-calonaci replied:
updated

SPPearce reviewed (commented):
Some comments when I had some time

elena-buscaroli committed:
updating logos

elena-buscaroli committed:
remove yml from modules conf

elena-buscaroli committed:
mc to local modules

elena-buscaroli committed:
update nf-core modules

elena-buscaroli committed:
mc to CITATIONS.md

elena-buscaroli committed:
updating the correct logos

nicola-calonaci committed:
Update CITATIONS.md

Co-authored-by: Simon Pearce <24893913+SPPearce@users.noreply.github.com>

nicola-calonaci committed:
Merge pull request #42 from caravagnalab/dev

Updating modules and minor changes

elena-buscaroli committed:
fixing minor errors and typos

elena-buscaroli committed:
Merge branch 'nf-core:dev' into dev

nicola-calonaci committed:
Merge pull request #43 from caravagnalab/dev

Fixing compilation error

SPPearce commented on nextflow.config:
```diff
@@ -1,147 +1,444 @@
+        ]
+    ]
+    homePage        = 'https://github.com/nf-core/tumourevo'
+    description     = """Analysis pipleine to model tumour clonal evolution from WGS data (driver annotation, quality control of copy number calls, subclonal and mutational signature deconvolution)"""
```
```suggestion
    description     = """Analysis pipeline to model tumour clonal evolution from WGS data (driver annotation, quality control of copy number calls, subclonal and mutational signature deconvolution)"""
```

valerianilucrezia replied:
Updated!

SPPearce commented on nextflow.config:
```diff
@@ -1,147 +1,444 @@
+    cnaqc_kernel_adjust = "1"
+    cnaqc_kde = "TRUE"
+    cnaqc_starting_state_subclonal_evolution = "1:1"
+    cnaqc_cluster_subclonal_CCF = "FALSE"
```
Suggestion
You have some parameters that have upper case parts to them. We generally use only `snake_case` for parameters.

SPPearce replied:
also `cnaqc_cutoff_QC_PASS`, `mobster_K`, `viber_K`, `sparsesignatures_K`

valerianilucrezia replied:
Changed all the parameters with upper case parts.

SPPearce commented on nextflow.config:
```diff
@@ -1,147 +1,444 @@
+    viber_alpha_0 = "1e-06"
+    viber_a_0 = "1"
+    viber_b_0 = "1"
+    viber_maxIter = "5000"
```
Think this one should definitely be changed though.
```suggestion
    viber_max_iter = "5000"
```

SPPearce commented on subworkflows/local/subclonal_deconvolution/main.nf:
```diff
@@ -0,0 +1,76 @@
+include { VIBER } from "../../../modules/nf-core/viber/main"
+include { PYCLONEVI } from "../../../modules/nf-core/pyclonevi/main"
+include { FORMATTER } from "../../../subworkflows/local/formatter/main"
+include { CTREE as CTREE_MOBSTERh } from "../../../modules/nf-core/ctree/main"
```
why lower case h at the end?

valerianilucrezia replied:
Removed

SPPearce commented on subworkflows/local/genome_interpreter/main.nf:
Can this be deleted?

valerianilucrezia replied:
Yes, done it!

SPPearce commented on subworkflows/local/lifter/main.nf:
Can you check the formatting on this, it looks weird. Uses a mix of . and | and I'm sure that the language server will complain about not having _ in the maps.

valerianilucrezia replied:
We uniform the formatting to only . use

SPPearce commented on subworkflows/local/download_cache_vep/main.nf:
Not sure why this is a subworkflow when it just wraps the module?

valerianilucrezia replied:
Thank you for the suggestion, we delete all subworkflows with only one module

nicola-calonaci replied:
Deleted = made them single modules

SPPearce commented on subworkflows/local/annotate_driver/main.nf:
This has a lot of commented out code

valerianilucrezia replied:
Removed

SPPearce commented on subworkflows/local/qc/main.nf:
We normally use lower case for file names like this

valerianilucrezia replied:
We changed it

SPPearce commented on subworkflows/local/qc/main.nf:
A lot of commented out code here

valerianilucrezia replied:
Removed!

SPPearce reviewed (commented):
Some comments

nf-core-bot committed:
Template update for nf-core/tools version 3.5.1

elena-buscaroli committed:
Template update for nf-core/tools version 3.5.1

elena-buscaroli committed:
Merge with new TEMPLATE version

elena-buscaroli committed:
fixing linting

nicola-calonaci committed:
Merge pull request #45 from caravagnalab/dev

TEMPLATE version update

valerianilucrezia committed:
remove old test + solve vep issue + remove subworkflow vep

valerianilucrezia committed:
solve test + remove subworkflows

valerianilucrezia committed:
minor

valerianilucrezia committed:
minor to test

valerianilucrezia committed:
update nf-test

valerianilucrezia committed:
minor test

valerianilucrezia committed:
try add maxForks

valerianilucrezia committed:
modify .github/workflows/nf-test.yml disk space

valerianilucrezia committed:
modify .github/workflows/nf-test.yml shards

valerianilucrezia committed:
modify .github/workflows/nf-test.yml shards for all

RaqManzano committed:
added support for strelka indels

valerianilucrezia committed:
modify .github/workflows/nf-test.yml remove disk large

valerianilucrezia committed:
remove sigprofiler from nf-test (make OOM error)

valerianilucrezia committed:
add ignore

valerianilucrezia committed:
solve snapshot error

valerianilucrezia committed:
ignore

valerianilucrezia committed:
modify git ignore

valerianilucrezia committed:
modify .prettierignore

valerianilucrezia committed:
remove unstable tests

nicola-calonaci committed:
Merge pull request #46 from caravagnalab/dev

Updates test and subworkflows

SPPearce commented on .nf-test.log:
Don't think this should be here ;) 
Or any of the other files under .nf-test.

valerianilucrezia replied:
removed, thank u!

nicola-calonaci committed:
Merge pull request #47 from RaqManzano/patch-strelka-indels

Added support for strelka indels in VCF2CNAQC

RaqManzano committed:
Merge pull request #1 from nf-core/dev

Updating dev from nf-core repo

RaqManzano committed:
Fixed sample names for strelka

RaqManzano committed:
Fixed vep config to avoid input/output naming conflict

RaqManzano committed:
Fixed error on indels logic

RaqManzano committed:
Adding cnaqc_genome param and removing cnaqc_plot_cn (not working)

RaqManzano committed:
Adding cnaqc_genome param and removing cnaqc_plot_cn (not working)

RaqManzano committed:
Changing cnaqc config format for args to make params work on the script.

RaqManzano committed:
Improved input staging

SPPearce commented on tests/nextflow.config:
```diff
@@ -0,0 +1,58 @@
+
+    max_cpus   = 4
+    max_memory = '15.GB'
+    max_time   = '4.h'
```
shouldn't need these any more?

valerianilucrezia replied:
yes, removed

RaqManzano committed:
prettier

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                },
+                "genome": {
+                    "type": "string",
+                    "description": "Reference genome name."
```
What is this used for? Are there fixed options?

valerianilucrezia replied:
this is used as a parameter for different tools like CNAqc, it is a required parameters otherwise the tools are not able to run

nicola-calonaci replied:
@valerianilucrezia is there any fixed option we may use?

valerianilucrezia replied:
yes, they are `["GRCh38", "GRCh37"]` and now we have make them explicit

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                    "type": "string",
+                    "default": "prior",
+                    "description": "Initialization of the q-distribution to compute the approximation of the posterior distributions.",
+                    "help_text": "This can be set in three different waysL equal to the prior (q_init = 'prior'), via kmeans clustering (q_init = 'kmeans') and capturing points which are private to each dimension (q_init = 'private')."
```
```suggestion
                    "help_text": "This can be set in three different ways: equal to the prior (q_init = 'prior'), via kmeans clustering (q_init = 'kmeans') and capturing points which are private to each dimension (q_init = 'private')."
```

SPPearce replied:
Maybe don't need the `q_init =` in these descriptions too.

valerianilucrezia replied:
done

nicola-calonaci replied:
@valerianilucrezia I think you left the `q_init` there, didn't you?

valerianilucrezia replied:
no, I've removed all of them

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "viber_re_assign": {
+                    "type": "string",
+                    "default": "FALSE",
+                    "description": "Boolean value whether point assigned to a cluster that is filtered our, are re-assigned from the density function."
```
```suggestion
                    "description": "Boolean value whether points assigned to a cluster that is filtered out, are re-assigned from the density function."
```

valerianilucrezia replied:
done

RaqManzano committed:
Updated snap due to new vep output name

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                    "type": "string",
+                    "default": "NULL",
+                    "description": "Deprecated parameter."
+                },
```
Can we remove this before the release then?

valerianilucrezia replied:
yes, done

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                    "type": "string",
+                    "default": "TRUE",
+                    "description": "Deprecated parameter."
+                },
```
And this one

nicola-calonaci replied:
done

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "publish_dir_mode_genome_sigprofiler": {
+                    "type": "string",
+                    "default": "move",
+                    "description": "Mode of publishing the SigProfiler genome."
```
What are the options for this? What does it do?

valerianilucrezia replied:
the options are [`"symlink", "rellink", "copy", "copyNoFollow", "move"`] and now they have been introduced in the `nextflow_schema.json`, thank you!

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "sigprofiler_nmf_replicates": {
+                    "type": "integer",
+                    "default": 100,
+                    "description": "The number of iteration to be performed to extract each number signature."
```
```suggestion
                    "description": "The number of iterations to be performed to extract each number signature."
```

nicola-calonaci replied:
done

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "sigprofiler_max_nmf_iterations": {
+                    "type": "integer",
+                    "default": 1000000,
+                    "description": "Value defines the maximum number of iterations to be completed before NMF converges ."
```
```suggestion
                    "description": "The maximum number of iterations to be completed before NMF converges."
```

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "sigprofiler_min_nmf_iterations": {
+                    "type": "integer",
+                    "default": 10000,
+                    "description": "Value defines the minimum number of iterations to be completed before NMF converges."
```
```suggestion
                    "description": "The minimum number of iterations to be completed before NMF converges."
```

nicola-calonaci replied:
done

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+                "sigprofiler_nmf_test_conv": {
+                    "type": "integer",
+                    "default": 10000,
+                    "description": "Value defines the number number of iterations to done between checking next convergence ."
```
```suggestion
                    "description": "The number of iterations to done between checking for convergence."
```

SPPearce commented on nextflow_schema.json:
```diff
@@ -0,0 +1,682 @@
+            "type": "string",
+            "default": "2:10"
+        }
+    }
```
Why is this bit here?

valerianilucrezia replied:
was a mistake, solved now, thank you

SPPearce commented on tests/default.nf.test:
You have just the one test? And it only covers two of the tools?

valerianilucrezia replied:
For now, yes, we’ve decided to develop only a single test, but this can easily be extended if you believe additional tests would be useful.
Regarding the tools, we encountered resource-related issues, specifically out-of-memory (OOM) errors for two tools (SparseSignature and SigProfiler). For VIBER, the module fails on the test dataset and the error is ignored; this is expected behavior, but it appears to be incompatible with the nf-test workflow.

SPPearce replied:
Well, ideally you'd be testing the basic functionality for all the tools.
Or at very least, a stub test.
Do you have a working dataset for them in the `test_full` config, so that it will be ran on the pipeline release?

SPPearce replied:
Note this isn't going to block the release, just a good idea.

valerianilucrezia replied:
we added a test for the stub for the moment, thank you for the suggestion!

SPPearce commented on docs/output.md:
```diff
@@ -1,43 +1,438 @@
+|   ├── SigProfiler
+│       └── <dataset>
+|   └── SparseSignatures
+│       └── <dataset>
```
You do have a mix of cases here, with snake_case and CamelCase

valerianilucrezia replied:
now it is correct

nicola-calonaci replied:
@valerianilucrezia please provide an explanation of why it is now correct

valerianilucrezia replied:
I've uniformed the output folders to be all lowercase

SPPearce commented on docs/output.md:
```diff
@@ -1,43 +1,438 @@
+
+## Lifter
+
+The Lifter subworkflow is an optional step and it is run when single sample VCF file are provided. When multiple samples from the same patient are provided, the user can specify either a single joint VCF file, containing variant calls from all tumor samples of the patient (see [joint variant calling](https://nf-co.re/sarek/3.4.2/parameters/#joint_mutect2)), or individual sample specific VCF files ...
```
"The Lifter subworkflow is an optional step and it is run when single sample VCF file are provided."
So each vcf file is a single sample?

valerianilucrezia replied:
yes, do you think we should rephrase it to make it clearer?

nicola-calonaci replied:
I suggest something like
: "The Lifter subworkflow is an optional step, and it is run when, despite having multiple samples per patient, variant calling is performed separately on each sample. In this case, instead of a VCF file with a column per sample (multi-sample variant calling), a single VCF file is provided per sample."

SPPearce replied:
Yep, that makes more sense to me.

SPPearce commented on docs/output.md:
```diff
@@ -1,43 +1,438 @@
+
+### TINC
+
+[TINC](https://caravagnalab.github.io/TINC/index.html) is a package to calculate the contamination of tumor DNA in a matched normal sample. TINC provides estimates of the proportion of cancer cells containing the normal sample (TIN), and the proportion of cancer cells in the tumor sample (TIT).
```
```
TINC provides estimates of the proportion of cancer cells containing the normal sample (TIN)
```
Not sure what exactly this means?

valerianilucrezia replied:
rewritten, hope it is clear now

nicola-calonaci replied:
@valerianilucrezia why don't we just use the description provided in the package webpage: "TINC is a package to determine the contamination of tumour DNA in a matched normal sample. The approach uses evolutionary theory applied to read counts data from whole-genome sequencing assays.

Precisely, the package provides methods to determine, for every matched pair of normal and tumour sample biopsies, the proportion of cancer cells, or tumour read fractions, contaminating the normal sample (Tumour in Normal, TIN). Similarly, it determines the proportion of cancer cells in the tumour sample (Tumour in Tumour, TIT), also called tumour purity.

If the TIN contamination determined by TINC is high, there are good chances that a somatic caller will be affected by false negative calls (i.e., non-called mutations that have read support in the normal). In these cases TINC can be used to flag samples better suited for a tumour-only somatic calling pipeline.". It sounds clearer
 to me.

valerianilucrezia replied:
I've substituted the old description with: TINC package provides methods to determine, for every matched pair of normal and tumour sample biopsies, the proportion of cancer cells, or tumour read fractions, contaminating the normal sample (Tumour in Normal, TIN). Similarly, it determines the proportion of cancer cells in the tumour sample (Tumour in Tumour, TIT), also called tumour purity.

SPPearce reviewed (commented):
I still need to check the rest of the outpud.md and usage.md

giorgiagandolfi committed:
set default label resources + add sigprofiler missing param

giorgiagandolfi committed:
add sigprofiler_seeds in nextflow_schema

valerianilucrezia committed:
Merge branch 'nf-core:dev' into dev

valerianilucrezia committed:
minor to doc + solve join position + minor test config

valerianilucrezia committed:
Merge branch 'dev' of github.com:caravagnalab/tumourevo into dev

valerianilucrezia committed:
minor to nextflow_schema

valerianilucrezia committed:
minor to output.md

valerianilucrezia committed:
Merge pull request #49 from caravagnalab/dev

Changes:
- set default label resources 
- add sigprofiler missing param
- remove .nf-test
- minor to nextflow_schema
- minor to output.md
- solve join_position when vcf is empty

SPPearce commented on docs/usage.md:
````diff
@@ -1,331 +1,319 @@
-```bash
-nextflow run nf-core/evoverse --reads '*_R{1,2}.fastq.gz' -profile docker
-```
+Through the analysis of variant and copy-number calls, it reconstructs the evolutionary process leading to the observed tumour genome. Most of the analyses can be done at mutliple levels: single sample, multiple samples from the same patient (multi-region/longitudinal assays), and multiple patients from distinct cohorts.
````
```suggestion
Through the analysis of variant and copy-number calls, it reconstructs the evolutionary process leading to the observed tumour genome. Most of the analyses can be done at multiple levels: single sample, multiple samples from the same patient (multi-region/longitudinal assays), and multiple patients from distinct cohorts.
```

valerianilucrezia replied:
Typo has been corrected

SPPearce commented on docs/usage.md:
```diff
@@ -1,331 +1,319 @@
+| `vcf` <br /> _Required_                    | Full path to the vcf file.                                                                                                                                                                                              |
+| `tbi` <br /> _Required_                    | Full path to the vcf `tabix` index file.                                                                                                                                                                                |
+| `cna_caller` <br /> _Required_             | Name of the copy number caller used to generate your data.                                                                                                                                                              |
+| `cna_segments` <br /> _Required_           | Full path to the segmentation files and copy number state from copy-number calling.                                                                                                                                     |
```
What are the options for this?

valerianilucrezia replied:
We have added the options both for cna_caller and vcf_caller, thank you!

valerianilucrezia committed:
nextflow_schema.json update

valerianilucrezia committed:
Merge branch 'dev' of github.com:caravagnalab/tumourevo into dev

valerianilucrezia committed:
add stub test and stub to missing modules

valerianilucrezia committed:
minor to test

valerianilucrezia committed:
improve lifter description

valerianilucrezia committed:
Merge pull request #50 from caravagnalab/dev

Add stub nf-test + minor

RaqManzano committed:
Update genome param

Changing genome param to global genome value

RaqManzano committed:
Update nextflow.config

Changing genome param to global genome value

RaqManzano committed:
Update nextflow_schema.json

Changing genome param to global genome value

valerianilucrezia committed:
Merge pull request #48 from RaqManzano/strelka-vep-cnaqc-patch

Strelka vep cnaqc patch

valerianilucrezia committed:
update snap

valerianilucrezia committed:
add yml test to stub

valerianilucrezia committed:
Merge pull request #51 from caravagnalab/dev

Update snapshot of tests

SPPearce commented on modules/local/get_positions/main_rel.nf:
```diff
@@ -0,0 +1,58 @@
+            dplyr::filter(chr %in% c(paste0('chr', seq(1,22)), 'chrX', 'chrY')) %>%
+            dplyr::select(chr, from)
+
+    write.table(file = paste0("$meta.tumour_sample", "_positions_missing"), missed, quote = F, sep = "\t", row.names = F, col.names = F)
```
This is a file called `foo_positions_missing`, with no extension?
But the emit calls it a `bed`, but I'm not sure it is in correct bed format.

valerianilucrezia replied:
now the module emits a bed file so we changed the file name to foo_positions_missing.bed, thank you!

SPPearce commented on docs/output.md:
```diff
@@ -1,43 +1,438 @@
+
+</details>
+
+## Lifter
```
Does this output any files?

valerianilucrezia replied:
yes, documentation is now updated

SPPearce commented on docs/output.md:
```diff
@@ -1,43 +1,438 @@
+
+</details>
+
+## QC
```
Does this output any files?

valerianilucrezia replied:
yes, documentation is now updated

SPPearce reviewed (commented):
Getting close now.

valerianilucrezia committed:
new round of review + new logic in qc

valerianilucrezia committed:
back to old logic + new figure + bed

kdavydzenka committed:
fix positions coordinates

valerianilucrezia committed:
Merge branch 'dev' of github.com:caravagnalab/tumourevo into dev

valerianilucrezia committed:
minort + new snapshot

valerianilucrezia committed:
Merge pull request #52 from caravagnalab/dev

Minor to docs + change qc logic

SPPearce reviewed (approved):
Ok, I'm happy now, ⭐

valerianilucrezia committed:
update mobster and viber + update workflow + minor error messages

valerianilucrezia committed:
Merge pull request #53 from caravagnalab/dev

Update mobster and viber + update workflow + minor error messages

mashehu commented on .nf-core.yml:
```diff
@@ -0,0 +1,19 @@
+lint:
+  files_exist:
+    - conf/modules.config
+  modules_config: false
```
```suggestion
```
I guess this is an outdated change, because modules.json is there now

mashehu commented on .nf-core.yml:
```diff
@@ -0,0 +1,19 @@
+  name: tumourevo
+  org: nf-core
+  outdir: .
+  version: 1.0.0
```
```suggestion
  version: 1.0.0
  skip_features:
    - multiqc
    - fastqc
```
https://nf-co.re/docs/nf-core-tools/pipelines/create#multiqc
will make future pipeline syncs easier.

mashehu reviewed (changes requested):
See comments  in #54

vvvirgy committed:
fixing typos

vvvirgy committed:
other typos fixed

vvvirgy committed:
schema and params typos

valerianilucrezia committed:
round of review

valerianilucrezia committed:
round of review 2

valerianilucrezia committed:
new review

valerianilucrezia committed:
new review

valerianilucrezia committed:
solve figure

valerianilucrezia committed:
create modules.config

valerianilucrezia committed:
update metromap

valerianilucrezia committed:
Merge pull request #56 from caravagnalab/dev

Minor docs and nextflow schema update

valerianilucrezia committed:
solve null in test.snap

valerianilucrezia committed:
minor to metromap

valerianilucrezia committed:
add output version.yml + minor to strelka parser

valerianilucrezia committed:
update version emission

valerianilucrezia committed:
update lifter

valerianilucrezia committed:
adjust VAF to lifted position

valerianilucrezia committed:
minor to doc

valerianilucrezia committed:
update last emission

valerianilucrezia committed:
update config

valerianilucrezia committed:
update mobster module

valerianilucrezia committed:
update test

valerianilucrezia committed:
minor to lifter

valerianilucrezia committed:
update versions.yml

valerianilucrezia committed:
update stub test

valerianilucrezia committed:
update stub test again

valerianilucrezia committed:
update stub test

valerianilucrezia committed:
update sparse signature + minor

valerianilucrezia committed:
Merge pull request #57 from caravagnalab/dev

- update test
- update metromap

mashehu reviewed (approved):
Tiny nit-pick you don't need to follow up for the release: https://github.com/nf-core/tumourevo/pull/54#discussion_r2983795405

Rest looks good now! Nice work!

valerianilucrezia committed:
update module ctree

valerianilucrezia committed:
Merge branch 'nf-core:dev' into dev

valerianilucrezia committed:
update channel version in subworkflow

valerianilucrezia committed:
update

valerianilucrezia committed:
Merge pull request #58 from caravagnalab/dev

Update module ctree

valerianilucrezia committed:
update filter sex chromosome viber and pyclone

valerianilucrezia committed:
try to solve test

valerianilucrezia committed:
clean nextflow config

valerianilucrezia committed:
remove patch to ctree

valerianilucrezia committed:
add again error_ignore to ctree

valerianilucrezia committed:
Merge pull request #60 from caravagnalab/dev

Update viber and pyclone: filter out sex chromosome

valerianilucrezia committed:
make it working with nextflow version 26.04

valerianilucrezia committed:
Merge pull request #62 from caravagnalab/dev

Update to work with nextflow version 26.04

valerianilucrezia committed:
update test volume

valerianilucrezia committed:
update test volume nf-test

valerianilucrezia committed:
Merge pull request #63 from caravagnalab/dev

Update test volume

valerianilucrezia committed:
update cnaqc and driver snapshot

valerianilucrezia committed:
update local modules

valerianilucrezia committed:
Merge branch 'nf-core:dev' into dev

valerianilucrezia committed:
Merge pull request #64 from caravagnalab/dev

Update conda environment.yml

valerianilucrezia committed:
adjust nexflow.config

valerianilucrezia committed:
Merge pull request #65 from caravagnalab/dev

Update test nexflow.config

valerianilucrezia committed:
minor to trigger checks

valerianilucrezia committed:
Merge branch 'dev' of github.com:caravagnalab/tumourevo into dev

valerianilucrezia committed:
Merge pull request #66 from caravagnalab/dev

minor to trigger checks

valerianilucrezia committed:
update tinc and pyclonevi modules

valerianilucrezia committed:
Merge branch 'nf-core:dev' into dev

valerianilucrezia committed:
adjust version tinc

valerianilucrezia committed:
Merge branch 'dev' of github.com:caravagnalab/tumourevo into dev

valerianilucrezia committed:
update stub test

valerianilucrezia committed:
add stub

valerianilucrezia committed:
Merge pull request #67 from caravagnalab/dev

Update tinc and pyclonevi modules

valerianilucrezia committed:
tumour only mode

valerianilucrezia committed:
update schema

valerianilucrezia committed:
adjust precommit

valerianilucrezia committed:
update

valerianilucrezia committed:
Merge pull request #68 from caravagnalab/dev

Add tumour only mode

valerianilucrezia committed:
minor to sequenza parser

valerianilucrezia committed:
update doc and metromap

valerianilucrezia committed:
Merge pull request #70 from caravagnalab/dev

Minor to sequenza parser

valerianilucrezia committed:
update platypus parser

valerianilucrezia committed:
Merge pull request #71 from caravagnalab/dev

Update platypus parser

valerianilucrezia committed:
update mobster + nextflow.config

valerianilucrezia committed:
update schema

valerianilucrezia committed:
update schema yml

valerianilucrezia committed:
update

valerianilucrezia committed:
update

nf-core-bot committed:
Remove vulnerable PR-comment artifact pattern

These files conflicted with the patch and were reset wholesale to the nf-core/tools 4.0.3 template version, on the basis that they carry no pipeline-specific content. If this branch had local edits in them, those edits are gone — please review: .github/workflows/branch.yml .github/workflows/template-version-comment.yml

mashehu committed:
Merge pull request #74 from nf-core/patch

Remove vulnerable PR-comment artifact pattern

valerianilucrezia committed:
Merge branch 'nf-core:dev' into dev

valerianilucrezia committed:
update mobster, pyclonevi, cnaqc, viber to work with new cnaqc indel and qc_chr logic

valerianilucrezia committed:
update schema

valerianilucrezia committed:
update modules.config

valerianilucrezia committed:
update modules.config

valerianilucrezia committed:
update snapshot

Commit:
updating cnaqc and join cnaqc with qc by chr

valerianilucrezia committed:
update mobster + joincnaqc + cnaqc

valerianilucrezia committed:
update joincnaqc

valerianilucrezia committed:
update gb for workflow action

valerianilucrezia committed:
update documentation and schema

valerianilucrezia committed:
Merge pull request #72 from caravagnalab/dev

Updates:
- mobster, pyclone, cnaqc modules
- add qc_chr flag
- add cnaqc_blacklist_indels flag
- add tumour-only mode
- update documentation
